CMakeLists.txt

```diff
@@ -33,6 +33,31 @@ option(DEBUG_INFO "Print message for developers to debug." OFF)
 option(ENABLE_NATIVE_OPTIMIZATION "Enable compilation optimization for the native machine's CPU type" OFF)
 # Do not use the new container by default.
 option(ENABLE_CONTAINER "Enable the new multi-device container." OFF)
+option(COMMIT_INFO "Print commit information" ON)
+
+#read version from VERSION file
+file(STRINGS "VERSION" VERSION)
+string(STRIP ${VERSION} VERSION)
+add_definitions("-DVERSION=\"${VERSION}\"")
+
+if(COMMIT_INFO)
+  # Get commit hash
+  execute_process(
+    COMMAND git log -1 --pretty=format:%h
```

Comment:
What if the user downloads the code from release page but not from the git repo? In this case it is unable to tell the version from git.